### PR TITLE
Add local tools notebook

### PR DESCRIPTION
**Pull request by giancastro**

### CODE REVIEW

**Review by TomAugspurger — COMMENTED** [review]
Thanks, a few inline comments for now.

**Comment by TomAugspurger** [comment]
Can you also merge master and check the new linting rules we added? https://github.com/microsoft/PlanetaryComputerExamples/blob/main/CONTRIBUTING.md#linting

File: tutorials/local-tools.ipynb [hunk 1]
```diff
@@ -0,0 +1,599 @@
...
+    "In this notebook, we'll demonstrate how to use the [Xarray-spatial](http://xarray-spatial.org/) local tools functions supported by [Numpy](https://numpy.org/). The spatial functions available are:\n",
+    "- Cell Statistics\n",
+    "- Combine\n",
+    "- Lesser Frequency\n",
+    "- Equal Frequency\n",
+    "- Greater Frequency\n",
+    "- Lowest Position\n",
+    "- Highest Position\n",
+    "- Popularity\n",
+    "- Rank"
```
[the remark this answers is not in the document]
Comment (giancastro) [reply tutorials/local-tools.ipynb]: done 21f8773.

File: tutorials/local-tools.ipynb [hunk 2]
```diff
@@ -0,0 +1,599 @@
...
+   "source": [
+    "### Cell Statistics"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "01ab159c",
+   "metadata": {},
+   "source": [
+    "`xrspatial.local.cell_stats` calculates statistics from a raster dataset on a cell-by-cell basis."
```
[the remark this answers is not in the document]
Comment (giancastro) [reply tutorials/local-tools.ipynb]: done d1e64d5.

File: tutorials/local-tools.ipynb [hunk 3]
```diff
@@ -0,0 +1,599 @@
...
+   "source": [
+    "### Combine"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "87c34a83",
+   "metadata": {},
+   "source": [
+    "`xrspatial.local.combine` combines data of a raster dataset, assigning a unique output value to each unique combination of raster values."
```
[the remark this answers is not in the document]
Comment (giancastro) [reply tutorials/local-tools.ipynb]: done ee7e9ac.

**Comment by giancastro** [comment]
> Can you also merge master and check the new linting rules we added? https://github.com/microsoft/PlanetaryComputerExamples/blob/main/CONTRIBUTING.md#linting

@TomAugspurger done d44a365.

**Comment by TomAugspurger** [comment]
Thanks. This looks good. I think the last thing would be a paragraph at the start that introduces local tools. IMO, this introduction should answer

1. What is local tools?
2. Why would I want to use local tools?

**Comment by giancastro** [comment]
> Thanks. This looks good. I think the last thing would be a paragraph at the start that introduces local tools. IMO, this introduction should answer
> 
> 1. What is local tools?
> 2. Why would I want to use local tools?

@TomAugspurger, done b3d1e8a.

File: tutorials/local-tools.ipynb [hunk 4]
```diff
@@ -0,0 +1,615 @@
...
+   "source": [
+    "## Exploring Xarray-spatial local tools functions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "32b5e1bf",
+   "metadata": {},
+   "source": [
+    "Local tools operate at the cell level, where values of the same cell from a set of input rasters are used to calculate the values of the cells at the output raster.\n",
```
Comment (TomAugspurger) [inline tutorials/local-tools.ipynb:16]:
```suggestion
    "Local tools operate at the cell level, where values with the same position from a set of input rasters are used to calculate the values of the cells at the output raster.\n",
```

File: tutorials/local-tools.ipynb [hunk 5]
```diff
@@ -0,0 +1,615 @@
...
+    "Some examples of the application of local tools are:\n",
+    "- Change over time: You can use Local Tools to identify places where, for instance, the land use or temperature changed over time.\n",
```
Comment (TomAugspurger) [inline tutorials/local-tools.ipynb:18]:
```suggestion
    "- Change over time: You can use local tools to identify places where a value like land use or temperature changed over time.\n",
```

File: tutorials/local-tools.ipynb [hunk 6]
```diff
@@ -0,0 +1,615 @@
...
+    "- Average overtime: With the local tools, it's possible to calculate the average value of indicators like rainfall or sunshine hours for each location of the area of interest.\n",
```
Comment (TomAugspurger) [inline tutorials/local-tools.ipynb:19]:
```suggestion
    "- Aggregate over time: You can use local tools to aggregate values over time, for example calculating the average rainfall or hours of sunshine for each cell.\n",
```

File: tutorials/local-tools.ipynb [hunk 7]
```diff
@@ -0,0 +1,615 @@
...
+    "- Lower overtime: It would be possible to identify places where the temperature was below a certain value at least once in a period of time."
```
Comment (TomAugspurger) [inline tutorials/local-tools.ipynb:20]:
```suggestion
    "- Threshold over time: You can use local tools to identify places where a value is above or below a specified threshold, for example where the temperature was below a 0 °C."
```

**Review by TomAugspurger — COMMENTED** [review]
Thanks. Is it fair to say that, local tools is typically used for time-series analysis? Is it exclusively used for time-series? What about something like comparing the outputs of a landcover classification model to the ground truth, to get something like a spatial confusion matrix?

**Comment by thuydotm** [comment]
Thanks for suggesting the example of having a confusion matrix. Would be interesting to have that function in our toolset. I think local tools can definitely be used for non time-series analysis. It depends on what the story we're telling. If we calculate a cost surface from a few layers, and each layer presents a different type of cost, their sum aggregate will not be time-series related.

**Comment by TomAugspurger** [comment]
Just to be clear, I'm not looking for a worked-out example of landcover & constructing a confusion matrix using local tools. I'm just looking for a single-sentence example of some non-timeseries application of local tools. So it could be something like

- Evaluate land-cover classification models: You can use local tools to compare the predictions of a land-cover classification to ground-truth labels.

If that is indeed a valid use-case for local-tools.

**Comment by giancastro** [comment]
@TomAugspurger, I added (abbb126) the use case that Thuy mentioned. What do you think?

**Comment by marcozimmermannpm** [comment]
> I'm not sure what a "cost surface" is. Is that a widely known term?

Yes @TomAugspurger, you can read more about cost surface here: http://wiki.gis.com/wiki/index.php/Cost_surface and here: https://pro.arcgis.com/en/pro-app/latest/tool-reference/spatial-analyst/creating-a-cost-surface-raster.htm

Should we try to make the text more clear for those who are not acquainted with the term?

**Comment by TomAugspurger** [comment]
Perhaps inlclude a link to the wikipedia page.

**Comment by TomAugspurger** [comment]
Thanks!

**Pull request merged**